tools/challenge-helper-scripts/create-project.ts

```diff
@@ -41,34 +47,136 @@ interface CreateProjectArgs {
   superBlock: SuperBlocks;
   block: string;
   helpCategory: string;
-  order: number;
+  blockType?: string;
+  blockLayout?: string;
+  questionCount?: number;
+  order?: number;
+  chapter?: string;
+  position?: number;
+  module?: string;
   title?: string;
 }
 
-async function createProject(
-  superBlock: SuperBlocks,
-  block: string,
-  helpCategory: string,
-  order: number,
-  title?: string
-) {
-  if (!title) {
-    title = block;
+async function createProject(projectArgs: CreateProjectArgs) {
+  if (!projectArgs.title) {
+    projectArgs.title = projectArgs.block;
   }
-  void updateIntroJson(superBlock, block, title);
 
-  const challengeId = await createFirstChallenge(superBlock, block);
-  void createMetaJson(block, title, helpCategory, challengeId);
+  const order = projectArgs.order!;
+  const chapter = projectArgs.chapter!;
+  const module = projectArgs.module!;
+
   const superblockFilename = (
     superBlockToFilename as Record<SuperBlocks, string>
-  )[superBlock];
-  // TODO: handle full-stack-developer (createProjects needs calling with a
-  // chapter and module name as well)
-  if (superBlock !== SuperBlocks.FullStackDeveloper) {
-    void updateSimpleSuperblockStructure(block, { order }, superblockFilename);
+  )[projectArgs.superBlock];
+
+  if (projectArgs.superBlock !== SuperBlocks.FullStackDeveloper) {
+    void updateSimpleSuperblockStructure(
+      projectArgs.block,
+      { order },
+      superblockFilename
+    );
+  } else {
+    void updateChapterModuleSuperblockStructure(
+      projectArgs.block,
+      { order, chapter, module },
+      superblockFilename
+    );
+  }
```

Comment:
```suggestion
  if (projectArgs.superBlock === SuperBlocks.FullStackDeveloper) {
    if (!chapter || !module || typeof projectArgs.position == 'undefined') {
      throw Error(
        'Missing one of the following arguments: chapter, module, position'
      );
    }
    void updateChapterModuleSuperblockStructure(
      projectArgs.block,
      { order: position, chapter, module },
      superblockFilename
    );
  } else {
    if (typeof order == 'undefined') {
      throw Error('Missing argument: order');
    }
    void updateSimpleSuperblockStructure(
      projectArgs.block,
      { order },
      superblockFilename
    );
  }
```
It's a bit awkward, but the validation can go here. Also, it's a bit easier to reason about if you avoid the negation.